pyproject.toml

```diff
@@ -84,3 +84,112 @@ required-imports = ["from __future__ import annotations"]
 [tool.ruff.pydocstyle]
 # Use Google-style docstrings.
 convention = "google"
+
+[project]
```

Comment:
can we put all the project specific config to the top of the file and all the tools at the bottom

Reply:
there you go.